community/kernel/src/main/java/org/neo4j/kernel/impl/util/DumpLogicalLog.java

```diff
@@ -88,43 +91,83 @@ public int dump( String filenameOrDirectory, String logPrefix, PrintStream out,
                 fileChannel.close();
                 throw ex;
             }
-            out.println( "Logical log format:" + logHeader.logFormatVersion + "version: " + logHeader.logVersion +
+            out.println( "Logical log format: " + logHeader.logFormatVersion + " version: " + logHeader.logVersion +
                     " with prev committed tx[" + logHeader.lastCommittedTxId + "]" );
 
-//            LogDeserializer deserializer = new LogDeserializer();
-
             PhysicalLogVersionedStoreChannel channel = new PhysicalLogVersionedStoreChannel(
                     fileChannel, logHeader.logVersion, logHeader.logFormatVersion );
             ReadableVersionableLogChannel logChannel =
                     new ReadAheadLogChannel( channel, NO_MORE_CHANNELS, 4096 );
 
-            try ( IOCursor<LogEntry> cursor = new LogEntryCursor( logChannel ) )
+            IOCursor<LogEntry> entryCursor = new LogEntryCursor( logChannel );
+            TransactionLogEntryCursor transactionCursor = new TransactionLogEntryCursor( entryCursor );
+            try ( IOCursor<LogEntry[]> cursor = regex == null ? transactionCursor :
+                    new FilteringIOCursor<>( transactionCursor, new TransactionRegexCriteria( regex ) ) )
```

Comment:
Good idea, I think it occurred as a thought during development, but as you said might be overkill a.t.m.